src/Avalonia.Controls/Platform/Screen.cs

```diff
@@ -60,6 +60,24 @@ public Screen(double scaling, PixelRect bounds, PixelRect workingArea, bool isPr
             this.Bounds = bounds;
             this.WorkingArea = workingArea;
             this.IsPrimary = isPrimary;
-        } 
+        }
+
+        /// <summary>
+        /// Determines whether the <see cref="Screen"/> is equal to the specified object.
+        /// </summary>
+        /// <param name="o">The object with which to test equality.</param>
+        /// <returns>True if the objects are equal, otherwise false.</returns>
+        public override bool Equals(object? o)
+        {
+            if (o is Screen screen)
+            {
+                return IsPrimary == screen.IsPrimary &&
+                       Bounds == screen.Bounds &&
+                       WorkingArea == screen.WorkingArea &&
+                       Math.Abs(Scaling - screen.Scaling) < double.Epsilon;
```

Comment:
Doesn't Avalonia have it's own Epsilon defined somewhere?

> Two apparently equivalent floating-point numbers might not compare equal because of differences in their least significant digits. For example, the C# expression, (double)1/3 == (double)0.33333, does not compare equal because the division operation on the left side has maximum precision while the constant on the right side is precise only to the specified digits. If you create a custom algorithm that determines whether two floating-point numbers can be considered equal, we do not recommend that you base your algorithm on the value of the [Epsilon](https://learn.microsoft.com/en-us/dotnet/api/system.double.epsilon?view=net-7.0#system-double-epsilon) constant to establish the acceptable absolute margin of difference for the two values to be considered equal. (Typically, that margin of difference is many times greater than [Epsilon](https://learn.microsoft.com/en-us/dotnet/api/system.double.epsilon?view=net-7.0#system-double-epsilon).) For information about comparing two double-precision floating-point values, see [Double](https://learn.microsoft.com/en-us/dotnet/api/system.double?view=net-7.0) and [Equals(Double)](https://learn.microsoft.com/en-us/dotnet/api/system.double.equals?view=net-7.0#system-double-equals(system-double)).

https://learn.microsoft.com/en-us/dotnet/api/system.double.epsilon?view=net-7.0

Reply:
Yeah tis been a while since i checked for the DoubleUtils class, should've used that